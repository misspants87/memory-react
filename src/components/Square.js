import React, { useEffect } from 'react'

const Square = props => {

let content  = () => {
    console.log(`square at index ${props.index} should show: ${props.showImage}`);
    if (props.showImage) {
      return (
        <img src={props.square} alt=""/>
      )
    } else {
      return null;
    }
  }


  return (
    <button onClick = {props.onClick.bind(this)}>
      {content()}
    </button>
  )
}

//if returns true, won't rerender
//prevProps.showImage === nextProps.showImage doing weird stuff. Should work?
export default React.memo(Square, (prevProps, nextProps) => {
  return (
    prevProps === nextProps
  )
});
