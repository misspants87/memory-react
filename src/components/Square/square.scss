button {
  background-color: white;
}

.flip-container .flipper {
  transform: rotateY(180deg);
}

.front, .back {
  width: 80px;
  height: 80px;
}

.flipper {
  transition: 0.6s;
  transform-style: preserve-3d;
  position: relative;
}
