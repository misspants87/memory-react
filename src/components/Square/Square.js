import React, { useEffect } from 'react';
import './square.scss';

const Square = props => {

let content  = () => {
    if (props.showImage) {
      return (
        <div className='flipper'>
          <div className='front'>
            <img src={props.square} alt=""/>
          </div>
        </div>
      )
    } else {
      return (
        <div className='back'>
        </div>
      );
    }
  }

  return (
    <div className='flip-container'>
      <button
      className='square-button'
      onClick = {props.onClick.bind(this)}
      data-test="Square Component">
        {content()}
      </button>
    </div>
  )
}

//if returns true, won't rerender
//prevProps.showImage === nextProps.showImage doing weird stuff. Should work? But it's not adding the second clickedSquare to exposedSquares... When Square isn't rerendering because I've added showImage to the comparison expression, it impacts exposedSquares of Game. The only square that still shows/hides when clicked (after the first and second click), is the second clicked square. Since that's the only one that's rerendered. It must be being passed back to the parent through onClick?
export default React.memo(Square, (prevProps, nextProps) => {
  console.log(prevProps.showImage === nextProps.showImage);
  return (
    prevProps === nextProps
  )
});
