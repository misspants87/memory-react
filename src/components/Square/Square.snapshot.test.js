import React from 'react';
import renderer from 'react-test-renderer';
import Square from './Square';

//snapshot tests just make sure that your UI doesn't change unexpectedly during development. Should also still write unit and functional tests, to verify that the application behaves as it should, not just that it hasn't changed
test('should render Square component correctly, with showImage prop equal to true as well as false', () => {
  const component = renderer.create(
    <Square
     showImage = {true}
     onClick = {() => 'fake callback'}
    />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.showImage = false
  expect(tree).toMatchSnapshot();
})
