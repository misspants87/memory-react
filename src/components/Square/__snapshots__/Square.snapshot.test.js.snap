// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Square component correctly, with showImage prop equal to true as well as false 1`] = `
<button
  data-test="Square Component"
  onClick={[Function]}
>
  <img
    alt=""
  />
</button>
`;

exports[`should render Square component correctly, with showImage prop equal to true as well as false 2`] = `
<button
  data-test="Square Component"
  onClick={[Function]}
  showImage={false}
>
  <img
    alt=""
  />
</button>
`;
